rs/boundary_node/anonymization/backend/src/registry.rs

```diff
@@ -0,0 +1,55 @@
+use anyhow::anyhow;
+use async_trait::async_trait;
+use candid::{CandidType, Principal};
+use serde::Deserialize;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ListError {
+    #[error(transparent)]
+    UnexpectedError(#[from] anyhow::Error),
+}
+
+#[async_trait]
+pub trait List: Sync + Send {
+    async fn list(&self) -> Result<Vec<Principal>, ListError>;
+}
+
+pub struct Client {
+    cid: Principal,
+}
+
+impl Client {
+    pub fn new(cid: Principal) -> Self {
+        Self { cid }
+    }
+}
+
+#[derive(CandidType)]
+pub struct EmptyStruct {}
```

Comment:
IMO, it's better to import this and `IdRecord` from [here](https://github.com/dfinity/ic/blob/2481ad5e85e5767b763fb3bf8eab2fe4c9a7636d/rs/registry/canister/src/gen/ic_registry_canister.pb.v1.rs#L135). I should do the same for the `rate-limit` canister.